File: 프로그래밍/py/code/03.단축평가.py
# ---
# check: none
# ---
total = 100
count = 0                        # 아직 아무도 응답하지 않았다

if count != 0 and total / count >= 60:
    print("평균이 60점 이상입니다")
else:
    print("아직 평균을 낼 수 없습니다")   # 이쪽이 실행된다

# count가 0이라 왼쪽이 거짓 -> 오른쪽 total / count는 «계산되지 않는다»
# 만약 계산했다면 0으로 나누어 그 자리에서 오류가 난다

File: 프로그래밍/c/code/03.단축평가.c
// ---
// check: none
// ---
int sum = 100;
int count = 0;                    // 아직 아무도 응답하지 않았다

if (count != 0 && sum / count >= 60) {
    printf("평균이 60점 이상입니다\n");
} else {
    printf("아직 평균을 낼 수 없습니다\n");   // 이쪽이 실행된다
}

// count가 0이라 왼쪽이 거짓 → 오른쪽 sum / count는 «계산되지 않는다»
// 만약 계산했다면 0으로 나누어 프로그램이 그 자리에서 죽는다
